File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/client/control/camera/ThirdPersonBackCameraTarget.java
package pize.tests.voxelgame.client.control.camera;

import pize.math.util.EulerAngles;
import pize.math.vecmath.vector.Vec3f;
import pize.tests.voxelgame.client.entity.LocalPlayer;

public class ThirdPersonBackCameraTarget implements CameraTarget{
    
    private final LocalPlayer targetPlayer;
    private final Vec3f position;
    private final EulerAngles direction;
    
    public ThirdPersonBackCameraTarget(LocalPlayer targetPlayer){
        this.targetPlayer = targetPlayer;
        position = new Vec3f();
        direction = new EulerAngles();
    }
    
    @Override
    public Vec3f getPosition(){
        final float dist = 5;
        position.set(targetPlayer.getLerpPosition())
            .add(0, targetPlayer.getEyeHeight(), 0)
            .add(targetPlayer.getRotation().getDirection().mul(-dist));
        return position;
    }
    
    @Override
    public EulerAngles getRotation(){
        direction.set(targetPlayer.getRotation());
        return direction;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/client/control/camera/ThirdPersonBackCameraTarget.kt
package pize.tests.voxelgame.client.control.camera

import pize.math.util.EulerAngles
import pize.math.vecmath.vector.Vec3f
import pize.tests.voxelgame.client.entity.LocalPlayer

class ThirdPersonBackCameraTarget(private val targetPlayer: LocalPlayer?) : CameraTarget {
    private override val position: Vec3f
    private val direction: EulerAngles

    init {
        position = Vec3f()
        direction = EulerAngles()
    }

    override fun getPosition(): Vec3f {
        val dist = 5f
        position.set(targetPlayer.getLerpPosition())
            .add(0f, targetPlayer.getEyeHeight(), 0f)
            .add(targetPlayer.getRotation().direction.mul(-dist))
        return position
    }

    override val rotation: EulerAngles
        get() {
            direction.set(targetPlayer.getRotation())
            return direction
        }
}
